tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,180 @@
+			})
+		})
+
+		When("default log level is used", func() {
+			It("only outputs info logs and the status is accepted and true", func() {
+				teardown(releaseName)
+
+				cfg := getDefaultSetupCfg()
+				cfg.infoLogLevel = true
+				setup(cfg)
+
+				ngfPodName, err := getNGFPodName()
+				Expect(err).ToNot(HaveOccurred())
+
+				nginxGateway := verifyAndReturnNginxGateway(nginxGatewayNsname)
+				Expect(nginxGateway.Status.Conditions[0].ObservedGeneration).To(Equal(int64(1)))
+
+				logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).ToNot(ContainSubstring("\"level\":\"debug\""))
+			})
+		})
+	})
+
+	When("testing on an existing NGF instance", Ordered, func() {
+		BeforeAll(func() {
+			var err error
+			ngfPodName, err = getNGFPodName()
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		When("NginxGateway is updated", func() {
+			It("captures the change, the status is accepted and true,"+
+				" and the observed generation is incremented", func() {
+				// previous test has left the log level at info, this test will change the log level to debug
+
+				var observedGeneration int64
+
+				nginxGateway := verifyAndReturnNginxGateway(nginxGatewayNsname)
+				observedGeneration = nginxGateway.Status.Conditions[0].ObservedGeneration
+
+				logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).ToNot(ContainSubstring("\"level\":\"debug\""))
+
+				Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+				// need to wait until files are applied, no current function because this is a nginx-gateway crd
+				time.Sleep(2 * time.Second)
+
+				nginxGateway = verifyAndReturnNginxGateway(nginxGatewayNsname)
+				Expect(nginxGateway.Status.Conditions[0].ObservedGeneration).To(Equal(observedGeneration + 1))
+
+				logs, err = resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).To(ContainSubstring(
+					"\"current\":\"debug\",\"msg\":\"Log level changed\",\"prev\":\"info\"",
+				))
+			})
+		})
+
+		When("NginxGateway is deleted", func() {
+			It("captures the deletion and default values are used", func() {
+				Expect(resourceManager.DeleteFromFiles(files, namespace)).To(Succeed())
+				time.Sleep(2 * time.Second) // need to wait until deletion is fully processed
+
+				_, err := getNginxGateway(nginxGatewayNsname)
+				Expect(err).Should(HaveOccurred())
+
+				logs, err = resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).To(ContainSubstring("NginxGateway configuration was deleted; using defaults"))
```

Comment:
Let's wrap this check in a retry as well. In this case, we also record an event, so let's add a check for that: https://github.com/nginxinc/nginx-gateway-fabric/blob/964877fc3a69cddaa72dce4ad5d18e53e5e958ec/internal/mode/static/config_updater.go#L48

Reply:
Oops thought that by checking the logs that would be checking the event.

I added the check of the actual event, however I checked the message, type, and reason, but not the involvedObject because I'm not sure how to do that with all the object conversions and such. Will the added change suffice?

Reply:
Yeah, looks good.